### PR TITLE
Usage: add gauge for panel/datasource/transformer types

### PR DESCRIPTION
**Pull request by ryantxu**
This PR uses the new search index to report on how often each:
* panel type
* datasource type
* transformer type 

is used across every panel.  http://localhost:3000/metrics. now has:

NOTE: this requires the [panelTitleSearch](https://github.com/grafana/grafana/blob/main/pkg/services/featuremgmt/registry.go#L70) feature flag to run

```
# HELP grafana_panel_datasource_usage indicates how many panels across all dashboards reference each datasource type
# TYPE grafana_panel_datasource_usage gauge
grafana_panel_datasource_usage{name="cloudwatch",orgId="1"} 8
grafana_panel_datasource_usage{name="elasticsearch",orgId="1"} 8
grafana_panel_datasource_usage{name="grafana-googlesheets-datasource",orgId="1"} 7
grafana_panel_datasource_usage{name="grafana-iot-roci-datasource",orgId="1"} 26
grafana_panel_datasource_usage{name="grafana-iot-sitewise-datasource",orgId="1"} 7
grafana_panel_datasource_usage{name="grafana-iot-twinmaker-datasource",orgId="1"} 3
grafana_panel_datasource_usage{name="grafana-mqtt-datasource",orgId="1"} 1
grafana_panel_datasource_usage{name="grafana-oas-datasource",orgId="1"} 15
grafana_panel_datasource_usage{name="grafana-plctag-datasource",orgId="1"} 15
grafana_panel_datasource_usage{name="grafana-raspberry-pi-datasource",orgId="1"} 1
grafana_panel_datasource_usage{name="grafana-roci-datasource",orgId="1"} 2
grafana_panel_datasource_usage{name="grafana-search-datasource",orgId="1"} 1
grafana_panel_datasource_usage{name="grafana-service-catalog-datasource",orgId="1"} 1
grafana_panel_datasource_usage{name="grafana-signal-datasource",orgId="1"} 11
grafana_panel_datasource_usage{name="grafana-timestream-datasource",orgId="1"} 43
grafana_panel_datasource_usage{name="graphite",orgId="1"} 2
grafana_panel_datasource_usage{name="influxdb",orgId="1"} 4
grafana_panel_datasource_usage{name="loki",orgId="1"} 5
grafana_panel_datasource_usage{name="mssql",orgId="1"} 37
grafana_panel_datasource_usage{name="mysql",orgId="1"} 35
grafana_panel_datasource_usage{name="opentsdb",orgId="1"} 4
grafana_panel_datasource_usage{name="postgres",orgId="1"} 47
grafana_panel_datasource_usage{name="prometheus",orgId="1"} 13
grafana_panel_datasource_usage{name="ryantxu-noaa-datasource",orgId="1"} 5
grafana_panel_datasource_usage{name="testdata",orgId="1"} 837
# HELP grafana_panel_transformer_usage indicates how many panels use each transformer type
# TYPE grafana_panel_transformer_usage gauge
grafana_panel_transformer_usage{name="calculateField",orgId="1"} 5
grafana_panel_transformer_usage{name="concatenate",orgId="1"} 1
grafana_panel_transformer_usage{name="configFromData",orgId="1"} 6
grafana_panel_transformer_usage{name="convertFieldType",orgId="1"} 1
grafana_panel_transformer_usage{name="extractFields",orgId="1"} 2
grafana_panel_transformer_usage{name="fieldLookup",orgId="1"} 1
grafana_panel_transformer_usage{name="fieldsFromJSON",orgId="1"} 1
grafana_panel_transformer_usage{name="filterByValue",orgId="1"} 1
grafana_panel_transformer_usage{name="filterFieldsByName",orgId="1"} 9
grafana_panel_transformer_usage{name="groupBy",orgId="1"} 1
grafana_panel_transformer_usage{name="histogram",orgId="1"} 3
grafana_panel_transformer_usage{name="labelsToFields",orgId="1"} 2
grafana_panel_transformer_usage{name="organize",orgId="1"} 18
grafana_panel_transformer_usage{name="prepareTimeSeries",orgId="1"} 2
grafana_panel_transformer_usage{name="reduce",orgId="1"} 11
grafana_panel_transformer_usage{name="renameByRegex",orgId="1"} 1
grafana_panel_transformer_usage{name="roci-register-links",orgId="1"} 8
grafana_panel_transformer_usage{name="rowsToFields",orgId="1"} 5
grafana_panel_transformer_usage{name="seriesToColumns",orgId="1"} 8
grafana_panel_transformer_usage{name="setGeometry",orgId="1"} 1
grafana_panel_transformer_usage{name="sortBy",orgId="1"} 3
grafana_panel_transformer_usage{name="spatial",orgId="1"} 5
grafana_panel_transformer_usage{name="twinmaker-register-links",orgId="1"} 1
# HELP grafana_panel_type_usage a metric indicating how many panels across all dashboards use each plugin panel type
# TYPE grafana_panel_type_usage gauge
grafana_panel_type_usage{name="alertlist",orgId="1"} 3
grafana_panel_type_usage{name="annolist",orgId="1"} 3
grafana_panel_type_usage{name="barchart",orgId="1"} 27
grafana_panel_type_usage{name="bargauge",orgId="1"} 62
grafana_panel_type_usage{name="candlestick",orgId="1"} 9
grafana_panel_type_usage{name="canvas",orgId="1"} 14
grafana_panel_type_usage{name="dashlist",orgId="1"} 13
grafana_panel_type_usage{name="debug",orgId="1"} 28
grafana_panel_type_usage{name="edge-drag-panel",orgId="1"} 2
grafana_panel_type_usage{name="edge-draw-panel",orgId="1"} 13
grafana_panel_type_usage{name="edge-form-panel",orgId="1"} 3
grafana_panel_type_usage{name="edge-form2-panel",orgId="1"} 3
grafana_panel_type_usage{name="form",orgId="1"} 2
grafana_panel_type_usage{name="gauge",orgId="1"} 57
grafana_panel_type_usage{name="geomap",orgId="1"} 71
grafana_panel_type_usage{name="grafana-iot-roci-debug-panel",orgId="1"} 5
grafana_panel_type_usage{name="grafana-iot-roci-layout-panel",orgId="1"} 2
grafana_panel_type_usage{name="grafana-iot-roci-scene-viewer-panel",orgId="1"} 6
grafana_panel_type_usage{name="grafana-iot-roci-sceneviewer-panel",orgId="1"} 2
grafana_panel_type_usage{name="grafana-iot-roci-sitewatch-player-panel",orgId="1"} 2
grafana_panel_type_usage{name="grafana-polystat-panel",orgId="1"} 3
grafana_panel_type_usage{name="grafana-singlestat-panel",orgId="1"} 7
grafana_panel_type_usage{name="grafana-worldmap-panel",orgId="1"} 5
grafana_panel_type_usage{name="graph",orgId="1"} 298
grafana_panel_type_usage{name="graph3",orgId="1"} 18
grafana_panel_type_usage{name="heatmap",orgId="1"} 13
grafana_panel_type_usage{name="heatmap-new",orgId="1"} 10
grafana_panel_type_usage{name="histogram",orgId="1"} 9
grafana_panel_type_usage{name="icon",orgId="1"} 3
grafana_panel_type_usage{name="live",orgId="1"} 17
grafana_panel_type_usage{name="logs",orgId="1"} 5
grafana_panel_type_usage{name="market-trend",orgId="1"} 3
grafana_panel_type_usage{name="natel-discrete-panel",orgId="1"} 7
grafana_panel_type_usage{name="natel-plotly-panel",orgId="1"} 5
grafana_panel_type_usage{name="piechart",orgId="1"} 12
grafana_panel_type_usage{name="row",orgId="1"} 39
grafana_panel_type_usage{name="ryantxu-ajax-panel",orgId="1"} 5
grafana_panel_type_usage{name="ryantxu-board-panel",orgId="1"} 3
grafana_panel_type_usage{name="singlestat",orgId="1"} 30
grafana_panel_type_usage{name="stat",orgId="1"} 60
grafana_panel_type_usage{name="state-timeline",orgId="1"} 18
grafana_panel_type_usage{name="status-history",orgId="1"} 5
grafana_panel_type_usage{name="table",orgId="1"} 153
grafana_panel_type_usage{name="table-old",orgId="1"} 3
grafana_panel_type_usage{name="text",orgId="1"} 55
grafana_panel_type_usage{name="timeline",orgId="1"} 5
grafana_panel_type_usage{name="timeseries",orgId="1"} 350
grafana_panel_type_usage{name="video",orgId="1"} 2
grafana_panel_type_usage{name="xychart",orgId="1"} 9
```

TODO:
- [x] Is this the correct naming convention?
- ~~[ ] Is there a better option to support usage across all orgIds?~~
     * we are now only reporting orgId=1, this is a good enough sample

### CODE REVIEW

File: pkg/services/searchV2/bluge.go [hunk 1]
```diff
@@ -145,6 +146,25 @@ func initBlugeIndex(dashboards []dashboard, logger log.Logger) (*bluge.Reader, e
 				AddField(bluge.NewKeywordField(documentFieldLocation, location).Aggregatable().StoreValue()).
 				AddField(bluge.NewKeywordField(documentFieldKind, string(entityKindPanel)).Aggregatable().StoreValue()) // likely want independent index for this
 
+			for _, xform := range panel.Transformer {
```
Comment (ryantxu) [inline pkg/services/searchV2/bluge.go:149]: @FZambia -- this part will need to be migrated to https://github.com/grafana/grafana/pull/48968

File: pkg/services/searchV2/usage.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+var (
+	infoPanelUsage = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name:      "panel_type_usage_info",
+		Help:      "info metric that exposes how often each panel type is used",
+		Namespace: "grafana",
+	}, []string{"name", "orgId"})
+
+	infoDatasourceUsage = promauto.NewGaugeVec(prometheus.GaugeOpts{
```
[the remark this answers is not in the document]
Comment (ryantxu) [reply pkg/services/searchV2/usage.go:27]: that tells you how many are *defined*, not how often they show up in dashboards.

**Comment by ryantxu** [comment]
@bergquist maybe this would be more clear it it was named:
```
grafana_panel_type_usage
grafana_panel_datasource_usage
grafana_panel_transformer_usage
```
Is the "info" suffix necessary?  it is not like the others I see that are always 1.

Also, is the orgId label a reasonable approach?  We do not have a single index with all orgs

**Comment by bergquist** [comment]
> Also, is the orgId label a reasonable approach? We do not have a single index with all orgs

I'd skip orgId. Wont matter in most cases and if someone uses orgs it would probably be too many time series :)

File: pkg/services/searchV2/usage.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bergquist) [reply pkg/services/searchV2/usage.go:27]: :man_facepalming:

**Review by bergquist — APPROVED** [review]
> Is this the correct naming convention?

yupp! :)

**Pull request merged**